contracts/staking/PoSValidatorManager.sol

```diff
@@ -130,4 +144,196 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
 
     function _lock(uint256 value) internal virtual returns (uint256);
     function _unlock(uint256 value, address to) internal virtual;
+
+    function _initializeDelegatorRegistration(
+        bytes32 validationID,
+        address delegator,
+        uint256 delegationAmount
+    ) internal nonReentrant {
+        uint64 weight = valueToWeight(_lock(delegationAmount));
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        // Ensure the validation period is active
+        Validator memory validator = _getValidator(validationID);
+        require(
+            validator.status == ValidatorStatus.Active, "PoSValidatorManager: validator not active"
+        );
+
+        // Ensure the delegator is not already registered
+        require(
+            $._delegatorStakes[validationID][delegator].weight == 0,
+            "PoSValidatorManager: delegator already registered"
+        );
+
+        _checkAndUpdateChurnTracker(weight);
+
+        // Update the validator weight
+        validator.weight += weight;
```

Comment:
I think we're going to need a new field to track the original weight of the validator. It'll be needed if the validator ends validation period to calculate their rewards, because right now we'd have to use the total weight with the delegations included.

Reply:
I went ahead and added that here, so that it doesn't fall through the cracks.

Reply:
I would advocate to only add it in when needed because future changes might not need it or want to implement differently, for example, whether `startTime` is only needed for PoS, or using a separate mechanism as a whole. Not a big deal though now that's already added.